python/lsst/ts/externalscripts/auxtel/latiss_acquire_and_take_sequence.py

```diff
@@ -240,116 +256,191 @@ def set_metadata(self, metadata):
+        if (self.acq_filter is not current_filter) and (self.acq_grating is not current_grating):
+            self.log.debug(f"Must load new filter {self.acq_filter } and grating {self.acq_grating}")
 
-        # Automatically accept calculated offset to sweetspot?
-        # if False this queries the user, which will fail badly on
-        # the script queue so make it always True
-        _alwaysAcceptMove = True
+            # Is the atspectrograph Correction in the ATAOS running?
+            corr = await self.atcs.rem.ataos.evt_correctionEnabled.aget(timeout=self.cmd_timeout)
+            if corr.atspectrograph:
+                # If so, then flush correction events for confirmation of
+                # corrections
+                self.atcs.rem.ataos.evt_atspectrographCorrectionStarted.flush()
+                self.atcs.rem.ataos.evt_atspectrographCorrectionCompleted.flush()
 
+        # Setup instrument and telescope
         await asyncio.gather(
-            self.attcs.slew_object(name=self.object_name, pa_ang=0, slew_timeout=240),
-            self.latiss.setup_atspec(grating=self.acq_grating, filter=self.acq_filter),
+            self.atcs.slew_object(name=self.object_name, rot=0, slew_timeout=240),
+            self.latiss.setup_instrument(grating=self.acq_grating, filter=self.acq_filter),
         )
 
-        success = False
-        iterNum = -1  # heinous to support max_acq_iter = 0 working TODO: fix this!
-        for iterNum in range(self.max_acq_iter):
-
-            retvals = await asyncio.gather(
-                findOffsetsAndMove(
-                    self.attcs,
-                    targetPosition,
-                    self.latiss,
-                    dataId=None,
-                    butler=self.butler,
-                    doMove=True,
-                    alwaysAcceptMove=_alwaysAcceptMove,
-                ),
+        # If ATAOS is running wait for adjustments to complete before
+        # moving on.
+        if corr.atspectrograph:
+            self.log.debug(f"Verifying LATISS configuration is incorporated into ATAOS offsets")
+            # If so, then flush correction events for confirmation of
+            # corrections
+            await self.atcs.rem.ataos.evt_atspectrographCorrectionStarted.aget(timeout=self.cmd_timeout)
+            await self.atcs.rem.ataos.evt_atspectrographCorrectionCompleted.aget(timeout=self.cmd_timeout)
+
+        self.log.info(
+            "Entering Acquisition Iterative Loop, with a maximum amount of "
+            f"iterations set to {self.max_acq_iter}"
+        )
+        iter_num = 0
+        while iter_num < self.max_acq_iter:
+            # Take image
+            self.log.debug(f"Starting iteration number {iter_num+1}")
+            tmp, data_id = await asyncio.gather(
                 self.latiss.take_object(exptime=self.acq_exposure_time, n=1),
+                get_next_image_dataId(self.latiss, timeout=self.acq_exposure_time + STD_TIMEOUT,),
```

Comment:
So, `take_object` returns the dataId, but seems like you want to wait for the `atarchiver.evt_imageInOODS` event. I think this could be incorporated to the latiss class. It could be an option on `take_object` for instance.

Reply:
So we could add it to `take_object`, but we will start slewing as soon as the image it done, but we can't do anything with the image until it's in the OODS.... so I think we need to keep these functionalities separate?
It's true that maybe `get_next_image_dataId` might not be capturing exactly what the method is doing (which is really saying when you can pull the image).

Reply:
uhm... not sure I understood your comment. What you mean by "we will start slewing as soon as the image is done"? Isn't this software performing an acquisition where you take an image, analyze it and perform offsets?

If you are talking in more general terms, my proposal is to add a new parameter, e.g. `wait_ingestion` that, if `True` would wait for the OODS event and return the data id. 

By default `wait_ingestion` would be `False` and the method retain the current behavior.

Reply:
So, it may be worth dealing with this particular feature in the future. Maybe just move `get_next_image_dataId` to be a method of this class and we implement it later in ts_observatory_control.

Reply:
ok.
You're correct about this software needing to wait for image analysis, but other use-cases certainly won't wait, which is why incorporating it into take_object doesn't address all use-cases...